### PR TITLE
Save average givback factor in donations

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to https://github.com/Giveth/giveth-dapps-v2/issues/1681#issuecomment-1326776980

### CODE REVIEW

**Review by CarlosQ96 — COMMENTED** [review]
Please rename all the methods everywhere for topPowerRank to bottomPowerRank, haha it was a bit confusing.

File: src/repositories/projectPowerViewRepository.test.ts [hunk 1]
```diff
@@ -470,7 +518,7 @@ function getTopPowerRankTestCases() {
     await setPowerRound(roundNumber);
     await refreshProjectPowerView();
 
-    const topPowerRank = await getTopPowerRank();
+    const topPowerRank = await getBottomRank();
```
Comment (CarlosQ96) [inline src/repositories/projectPowerViewRepository.test.ts]: rename variable. I understand now better what you meant, I was the one that named it wrongly.

File: src/resolvers/powerBoostingResolver.ts [hunk 2]
```diff
@@ -168,6 +168,6 @@ export class PowerBoostingResolver {
 
   @Query(returns => Number)
   async getTopPowerRank(): Promise<Number> {
-    return getTopPowerRank();
```
Comment (CarlosQ96) [inline src/resolvers/powerBoostingResolver.ts:171]: renameQuery

File: src/repositories/projectPowerViewRepository.ts [hunk 3]
```diff
@@ -9,14 +9,21 @@ export const getProjectPowers = async (
...
+
 export const getProjectFuturePowers = async (
   take: number = 50,
   skip: number = 0,
 ): Promise<ProjectPowerView[]> => {
   return ProjectFuturePowerView.find({ take, skip });
 };
 
-export const getTopPowerRank = async (): Promise<number> => {
+export const getBottomRank = async (): Promise<number> => {
```
Comment (CarlosQ96) [inline src/repositories/projectPowerViewRepository.ts:26]: Rename logs to bottom powerRank

File: src/resolvers/powerBoostingResolver.ts (hunk 2 above)
Comment (CarlosQ96) [reply src/resolvers/powerBoostingResolver.ts:171]: Unless frontend already implemented I think its ok to leave it like this for now. But I leave it up to you.

**Review by CarlosQ96 — APPROVED** [review]
Hey Mohammad LGTM. Just rename the topPowerRank to bottomPowerRank everywhere so its not confusing.

**Comment by CarlosQ96** [comment]
Fixed merge conflicts considering the onramper refactor. Double check just in case. but it seems to be ok.

File: src/repositories/projectPowerViewRepository.ts (hunk 3 above)
Comment (mohammadranjbarz) [reply src/repositories/projectPowerViewRepository.ts:26]: Thanks you

File: src/repositories/projectPowerViewRepository.test.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply src/repositories/projectPowerViewRepository.test.ts]: It confused Mitch and suggested this new name haahaa

**Comment by mohammadranjbarz** [comment]
Thanks @CarlosQ96  for reviewing and improving naming

File: src/resolvers/powerBoostingResolver.ts (hunk 2 above)
Comment (mohammadranjbarz) [reply src/resolvers/powerBoostingResolver.ts:171]: Yeah I afraid of that, so I didnt change the name of query

**Pull request merged**